capstone/capweb/views.py

```diff
@@ -373,3 +375,17 @@ def download_files(request, filepath=""):
             "error": "This file was not found in our system."
         }
         return render(request, "file_download_400.html", context, status=404)
+
+
+def download_contents_file(request, filepath=""):
+    absolute_path = download_files_storage.path(filepath)
+    with tempfile.TemporaryFile() as tf:
```

Comment:
Rather than writing to a temp file, I would use the csv writer here with StringIO to avoid encoding problems of building your own string or the need to write out to disk: https://stackoverflow.com/a/9157370/307769 

For the csv contents, let's add column headers and have three columns for the path, file size, and modification date. This file will be cached so I think the cpu hit should be ok.